subsys/bluetooth/mesh/rpl.c

```diff
@@ -215,15 +217,25 @@ void bt_mesh_rpl_reset(void)
 				} else {
 					(void)memset(rpl, 0, sizeof(*rpl));
 				}
+
+				shift++;
 			} else {
 				rpl->old_iv = true;
 
 				if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
 					schedule_rpl_store(rpl, true);
 				}
+
+				if (shift > 0) {
+					replay_list[i - shift] = *rpl;
+				}
 			}
+
+			last = i;
 		}
 	}
+
+	memset(&replay_list[last - shift + 1], 0, sizeof(struct bt_mesh_rpl) * shift);
```

Comment:
```suggestion
	memset(&replay_list[ARRAY_SIZE(repaly_list) - shift], 0, sizeof(struct bt_mesh_rpl) * shift);
```
Is last always equal to ARRAY_SIZE(replay_list)-1, if so is last needed?

Reply:
Not always. If there are less stored address in RPL than the total amount of entries, then `last` will be lower than `ARRAY_SIZE(replay_list)-1`. `ARRAY_SIZE(replay_list)-1` is the worst case.

Reply:
This does not take into consideration that there maybe empty slots. Entries will not be move into the empty slots. The empty slots can be created by bt_mesh.flags != BT_MESH_VALID. (line:351) The empty slot will continue to be empty. We need to do the same there.
Should also (void) return value.